py/tests/ctm_test.py

```diff
@@ -22,44 +23,66 @@ def ctm_to_blob(ctm, card):
     return pykms.Blob(card, arr);
 
 
-if len(sys.argv) > 1:
-    conn_name = sys.argv[1]
-else:
-    conn_name = ""
+parser = argparse.ArgumentParser(description='Simple CRTC CTM-property test.')
+parser.add_argument('--connector', '-c', dest='connector',
+		    required=False, help='connector to output')
+parser.add_argument('--mode', '-m', dest='modename',
+		    required=False, help='Video mode name to use')
+parser.add_argument('--plane', '-p', dest='plane', type=int,
+		    required=False, help='plane number to use')
+args = parser.parse_args()
 
 card = pykms.Card()
 res = pykms.ResourceManager(card)
-conn = res.reserve_connector(conn_name)
+conn = res.reserve_connector(args.connector)
 crtc = res.reserve_crtc(conn)
-mode = conn.get_default_mode()
+format = pykms.PixelFormat.ARGB8888
+if args.modename == None:
+    mode = conn.get_default_mode()
+else:
+    mode = conn.get_mode(args.modename)
+modeb = mode.to_blob(card)
 
-fb = pykms.DumbFramebuffer(card, mode.hdisplay, mode.vdisplay, "XR24");
+fb = pykms.DumbFramebuffer(card, mode.hdisplay, mode.vdisplay, "AR24");
```

Comment:
Why AR24?

Reply:
Oh that is an obsolete test. Back when I implemented CTM for tidss the driver did not support XR24, and I did not want to do a context switch for that. I'll revert that back.